src/main/java/ca/corefacility/bioinformatics/irida/service/impl/sample/SampleServiceImpl.java

```diff
@@ -650,6 +646,19 @@ public List<QCEntry> getQCEntriesForSample(Sample sample) {
 		return qcEntryRepository.getQCEntriesForSample(sample);
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Transactional(readOnly = true)
+	@Override
+	@PreAuthorize("hasRole('ROLE_ADMIN') or hasPermission(#samples, 'canReadSample')")
+	public Map<Long, List<QCEntry>> getQCEntriesForSamples(List<Sample> samples) {
+		return qcEntryRepository.getQCEntriesForSamples(samples)
+				.stream()
+				.collect(Collectors.groupingBy(t -> (Long) t.get(0),
```

Comment:
Can we update `t` to something more meaningful so that it's clear what it is?

Reply:
Updated in cee5f10